build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/transport/TransportVersionValidationFuncTest.groovy

```diff
@@ -198,4 +198,35 @@ class TransportVersionValidationFuncTest extends AbstractTransportVersionFuncTes
         then:
         result.task(":myserver:validateTransportVersionDefinitions").outcome == TaskOutcome.SUCCESS
     }
+
+    def "latest can refer to an unreferenced definition"() {
+        given:
+        unreferencedTransportVersion("initial_10.0.0", "10000000")
+        latestTransportVersion("10.0", "initial_10.0.0", "10000000")
+        when:
+        def result = gradleRunner(":myserver:validateTransportVersionDefinitions").build()
+        then:
+        result.task(":myserver:validateTransportVersionDefinitions").outcome == TaskOutcome.SUCCESS
+    }
+
+    def "named and unreferenced definitions cannot have the same name"() {
+        given:
+        unreferencedTransportVersion("existing_92", "10000000")
+        when:
+        def result = validateDefinitionsFails()
+        then:
+        assertDefinitionsFailure(result, "Transport version definition file " +
+                "[myserver/src/main/resources/transport/definitions/named/existing_92.csv] " +
+                "has same name as unreferenced definition " +
+                "[myserver/src/main/resources/transport/definitions/unreferenced/existing_92.csv]")
+    }
+
+    def "unreferenced definitions can have primary ids that are patches"() {
```

Comment:
I'm curious, why?  These are created when a new branch is cut, which will result in a "base id" without a patch section, right?  Is there another scenario where a patch is needed?

Reply:
Yes. We bump the transport version on every release. So when we release 9.2.0 and bump the version to 9.2.1, we also bump the transport version, which in that case would be a patch version. This is to ensure that every stack version has a unique transport version.